src/constbuffer_array.c

```diff
@@ -857,6 +857,94 @@ IMPLEMENT_MOCKABLE_FUNCTION(, const CONSTBUFFER_HANDLE*, constbuffer_array_get_c
     return result;
 }
 
+IMPLEMENT_MOCKABLE_FUNCTION(, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_remove_empty_buffers, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_handle)
+{
+    CONSTBUFFER_ARRAY_HANDLE result;
+
+    if (constbuffer_array_handle == NULL)
+    {
+        /*Codes_SRS_CONSTBUFFER_ARRAY_88_001: [ If constbuffer_array_handle is NULL then constbuffer_array_remove_empty_buffers shall fail and return NULL. ]*/
+        LogError("Invalid arguments: CONSTBUFFER_ARRAY_HANDLE constbuffer_array_handle=%p", constbuffer_array_handle);
+        result = NULL;
+    }
+    else
+    {
+        /*Codes_SRS_CONSTBUFFER_ARRAY_88_002: [ constbuffer_array_remove_empty_buffers shall get the buffer count from constbuffer_array_handle. ]*/
+        if (constbuffer_array_handle->nBuffers == 0)
+        {
```

Comment:
I think we should not treat differently the case when the input const buffer array has 0 buffers. If we have 0 buffers we would figure out after examining the input const buffer array that the number of non empty buffers has not changed and we should simply return the same input const buffer array with its reference count incremented.